go/extractor/cli/go-autobuilder/go-autobuilder.go

```diff
@@ -58,14 +58,38 @@ Build behavior:
 
 var goVersion = ""
 
+// Returns the file system root. That is "/" on Unix-like systems and e.g. "C:" on Windows.
+func getFSRoot() string {
+	if runtime.GOOS == "windows" {
+		return os.Getenv("SystemDrive")
+	}
+	return "/"
+}
+
 // Returns the current Go version as returned by 'go version', e.g. go1.14.4
 func getEnvGoVersion() string {
 	if goVersion == "" {
-		gover, err := exec.Command("go", "version").CombinedOutput()
+		// Since Go 1.21, running 'go version' in a directory with a 'go.mod' file will attempt to
```

Comment:
This can be done more cleanly by setting environment variable `GOTOOLCHAIN=local`. That means the `go` command should run its bundled toolchain; most commands would choke on a `go.mod` demanding a higher version when forbidden to auto-upgrade in this way, but `GOTOOLCHAIN=local go version` reports the builtin toolchain's version as you might hope.